cmd/statusd/server/parse-api/main.go

```diff
@@ -0,0 +1,157 @@
+//go:generate go run main.go
+
+/*
+This script generates a Go file with a list of supported endpoints based on the public functions in `mobile/status.go`.
+The output has 3 sections:
+- Endpoints with a response of type `string`
+- Endpoints with no arguments and a response of type `string`
+- Unsupported endpoints: those have non-standard signatures
+Deprecated functions are ignored.
+*/
+
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+	"text/template"
+)
+
+const (
+	inputFilePath    = "../../../../mobile/status.go"
+	templateFilePath = "./endpoints_template.txt"
+	outputFilePath   = "../endpoints.go"
+)
+
+var (
+	// Regular expressions extracted as global variables
+	publicFunc                = regexp.MustCompile(`func\s+([A-Z]\w+)\(.*\).*\{`)
+	publicFuncWithRespPattern = regexp.MustCompile(`^func\s+([A-Z]\w*)\((\w|\s)+\)\s+string\s+\{$`)
+	publicFuncNoArgsPattern   = regexp.MustCompile(`^func\s+([A-Z]\w*)\(\)\s+string\s+\{$`)
+	funcNamePattern           = regexp.MustCompile(`^func\s+([A-Z]\w*)\(`)
+)
+
+type TemplateData struct {
+	PackageName          string
+	FunctionsWithResp    []string
+	FunctionsNoArgs      []string
+	UnsupportedEndpoints []string
+}
+
+func main() {
+	// Open the Go source file
+	file, err := os.Open(inputFilePath)
+	if err != nil {
+		fmt.Printf("Failed to open file: %s\n", err)
+		return
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	var publicFunctionsWithResp []string
+	var publicFunctionsNoArgs []string
+	var unsupportedFunctions []string
+	var isDeprecated bool
+	var packageName string
+
+	for scanner.Scan() {
+		line := strings.TrimSpace(scanner.Text())
+
+		// Detect package name
+		if strings.HasPrefix(line, "package ") {
+			packageName = strings.TrimPrefix(line, "package ")
+		}
+
+		// Check for deprecation comment
+		if isDeprecatedComment(line) {
+			isDeprecated = true
+			continue
+		}
+
+		if !publicFunc.MatchString(line) {
+			continue
+		}
+
+		if isDeprecated {
+			isDeprecated = false
```

Comment:
We should support deprecated by default and maybe add a switch to disable it. Consider that the mobile client will use the server one day. That would be a really cool experience for mobile developers as they won't need to endure the long wait to re-build with status-go with each modification to status-go to debug or do something else ...

Reply:
Makes sense indeed, will do 👍

Reply:
@qfrank, done: 181e58b46f6840363cd45682b6cf7ec9701ca342

Deprecated methods are now available, but will have `Deprecation: true` header 🙌 

<img width="477" alt="Screenshot 2024-09-26 at 15 32 49" src="https://github.com/user-attachments/assets/b47a00ff-21ce-4bb9-8b42-ca1d444ba982">